nixos/doc/manual/release-notes/rl-2211.section.md

```diff
@@ -179,6 +181,8 @@ Use `configure.packages` instead.
 
 - memtest86+ was updated from 5.00-coreboot-002 to 6.00-beta2. It is now the upstream version from https://www.memtest.org/, as coreboot's fork is no longer available.
 
+- Option descriptions, examples, and defaults writting in DocBook are now deprecated and will generate informational messages during the documentation build. Using CommonMark is preferred and will become the default in 23.05.
```

Comment:
This is also too fast. There needs to be a reasonable period for third-party tools to easily support multiple nixpkgs releases without causing warnings. This was previously a problem with iirc home-manager, where a deprecation warning was unavoidable due to older nixpkgs still needing support.

Reply:
> anything that wants to support both can either go the trivial route with a let mdDoc = lib.mdDoc or lib.id; in prefix and using documentation that works with both renderers, or it can have two texts and switch between them based on pkgs.version

Both of those are far from ideal and require a lot of work. Third-party tools should also have a chance for a smooth transition with an appropriate deprecation period. Merging this PR as-is would require home-manager, nix-darwin (and anything else rendering docs using the module system) to effectively immediately update all their docbook to markdown, support for which was only introduced [a couple months ago](https://github.com/NixOS/nixpkgs/pull/176146).

This is how I think the deprecation strategy should look like:
- 22.11: Introduce support for markdown docs (https://github.com/NixOS/nixpkgs/pull/176146). Mention in the release notes that support for docbook will be deprecated in the next release and removed in the one after that. No warning for docbook yet. This allows third-party tools to read release notes and prepare for the next release, such that they won't cause any warnings for users.
- 23.05: Deprecate docbook support by adding a warning when `mdDoc` isn't used. The warning mentions that docbook support will be removed in the next released. This pushes everybody to use `mdDoc`, explicitly confirming that they have transitioned their code to markdown.
- 23.11: Remove docbook support either by using markdown implicitly for unannotated strings. This is only safe assuming everybody did the transition to 23.05, if somebody updates from older releases, they'd never get the warning to move to `mdDoc`, but considering the alternative of throwing an error for unannotated strings, that's a worthwhile tradeoff imo.
- (future): Slowly deprecate `mdDoc` with a warning, then an error, then a complete removal


So essentially there's two deprecation phases: An initial silent one for third-party tools to prepare a smooth NixOS/nixpkgs upgrade for their users, and a noisy one (warning) for end users. And if a third-party tool misses the first phase of deprecation, they will get complaints by their users during the second one.

Reply:
Found the issue I was thinking of that shows how this problem happened in the past: https://github.com/NixOS/nixpkgs/pull/163451#event-6275878451